crates/bevy_render/src/view/window.rs

```diff
@@ -215,8 +229,28 @@ pub fn prepare_windows(
                 CompositeAlphaMode::PostMultiplied => wgpu::CompositeAlphaMode::PostMultiplied,
                 CompositeAlphaMode::Inherit => wgpu::CompositeAlphaMode::Inherit,
             },
+            // TODO: Use an sRGB view format here on platforms that don't support sRGB surfaces. (afaik only WebGPU)
+            view_formats: vec![],
         };
 
+        // This is an ugly hack to work around drivers that don't support MSAA.
+        // This should be removed once https://github.com/bevyengine/bevy/issues/7194 lands and we're doing proper
+        // feature detection for MSAA.
+        let sample_flags = render_adapter
+            .get_texture_format_features(surface_configuration.format)
+            .flags;
+        match *msaa {
+            Msaa::Off => (),
+            Msaa::Sample4 => {
+                if !sample_flags.contains(wgpu::TextureFormatFeatureFlags::MULTISAMPLE_X4) {
+                    bevy_log::warn!(
+                        "MSAA 4x is not supported on this device. Falling back to disabling MSAA."
+                    );
+                    *msaa = Msaa::Off;
+                }
+            }
+        }
```

Comment:
One problem I see with this is that MSAA can be toggled at runtime. We have an `msaa` example that tests that IIRC. What happens in that case? If it was a problem before wgpu 0.15 and so is not a regression then we can just create an issue and not block this PR.

Reply:
Pre 0.15 it wouldn't cause a panic, it just wouldn't work and would render strangely (weird white outlines around stuff, that no longer happens with this according to Jasmine (@JMS55)).

In 0.15 it panics.

@JMS55 tested this and it works both when MSAA is enabled at the start, and when it's toggled at runtime.

Reply:
To be clear, in 0.15 it panics when MSAA is turned on.

This code to disable MSAA works well and prevents the crash, including when toggled at runtime.